### PR TITLE
4673 Command line reports 

### PR DESCRIPTION
**Pull request by eugene7646**

### CODE REVIEW

File: Core/src/org/sleuthkit/autopsy/modules/case_uco/CaseUcoFormatExporter.java [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.autopsy.coreutils.MessageNotifyUtil;
+import org.sleuthkit.autopsy.datamodel.ContentUtils;
+import org.sleuthkit.autopsy.ingest.IngestManager;
+import org.sleuthkit.autopsy.report.ReportProgressPanel;
+import org.sleuthkit.datamodel.SleuthkitCase;
+import org.sleuthkit.datamodel.TskCoreException;
+import org.sleuthkit.datamodel.TskData;
+
+public class CaseUcoFormatExporter {
```
Comment (dgrove727) [inline Core/src/org/sleuthkit/autopsy/modules/case_uco/CaseUcoFormatExporter.java]: I realize a vast majority of the code here is essentially copy/paste, but somehow the Javadocs got left out off of the class and methods within. I would at least put these back in.

Comment (dgrove727) [reply Core/src/org/sleuthkit/autopsy/modules/case_uco/CaseUcoFormatExporter.java]: I also notice a few Codacy things that are probably easy to address.

File: Core/src/org/sleuthkit/autopsy/commandline/CommandLineIngestManager.java [hunk 2]
```diff
@@ -0,0 +1,602 @@
...
+                }
+
+                // read options panel configuration
+                String rootOutputDir = UserPreferences.getCommandLineModeResultsFolder();
+                if (!(new File(rootOutputDir).exists())) {
+                    LOGGER.log(Level.SEVERE, "The output directory doesn't exist {0}", rootOutputDir);
+                    System.out.println("The output directory doesn't exist " + rootOutputDir);
+                    return;
+                }
+                rootOutputDirectory = Paths.get(rootOutputDir);
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/commandline/CommandLineIngestManager.java]: Given that we write the case name and data source path to the log and to the console, shouldn't we also write the output directory path as well?

File: Core/src/org/sleuthkit/autopsy/commandline/CommandLineIngestManager.java [hunk 3]
```diff
@@ -0,0 +1,602 @@
...
+
+                // open case
+                Case caseForJob;
+                try {
+                    caseForJob = openCase(baseCaseName);
+                } catch (CaseActionException ex) {
+                    LOGGER.log(Level.SEVERE, "Error creating or opening case " + baseCaseName, ex);
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/commandline/CommandLineIngestManager.java]: Up to this point, we are writing to both the log and to System.out. After this we are just writing to the log. If we are not going to continue writing to both, it seems that we should at least have some sort of outcome message written to the console. Thoughts?

If we were going to continue to log everything in both places, one approach would be to write some private helper methods to write to both the log and the console.

Comment (eugene7646) [reply Core/src/org/sleuthkit/autopsy/commandline/CommandLineIngestManager.java]: I have added the System.out messages to all logging in the JobProcessingTask.run() method. My thought process was that System.out are there to help the user debug initial configuration and input parameter errors. I think for debugging "bigger" issues, like errors during ingest or in DSPs, we should continue to rely on application logs, so I think limiting System.out  logging to only the main JobProcessingTask.run() method is sufficient. @rcordovano Agree?

Comment (rcordovano) [reply Core/src/org/sleuthkit/autopsy/commandline/CommandLineIngestManager.java]: Sure. I do think we should add some sort of completion message to the console when the task is done.

Comment (eugene7646) [reply Core/src/org/sleuthkit/autopsy/commandline/CommandLineIngestManager.java]: Oh yes, I have duplicated all logging in JobProcessingTask.run(), to have both application log and System.out, including task completion. I just haven't pushed up my changes yet.

File: Core/src/org/sleuthkit/autopsy/commandline/CommandLineIngestManager.java [hunk 4]
```diff
@@ -0,0 +1,602 @@
...
+
+    public void start() {
+        JobProcessingTask jobProcessingTask = new JobProcessingTask();
+        jobProcessingTaskFuture = jobProcessingExecutor.submit(jobProcessingTask);
+    }
+
+    public void stop() {
+
+        jobProcessingTaskFuture.cancel(true);
+        jobProcessingExecutor.shutdown();
```
[the remark this answers is not in the document]
Comment (rcordovano) [reply Core/src/org/sleuthkit/autopsy/commandline/CommandLineIngestManager.java]: Okay, so we are going to go with an exceptions firewall instead, for now.

**Pull request merged**